node/metrics/createQuote.ts

```diff
@@ -0,0 +1,106 @@
+import type { Metric } from './metrics'
+import { sendMetric } from './metrics'
+
+type UserData = {
+  orgId: string
+  costId: string
+  roleId: string
+}
+
+type SessionData = {
+  namespaces: {
+    profile: {
+      id: { value: string }
+      email: { value: string }
+    }
+    account: {
+      accountName: { value: string }
+    }
+  }
+}
+
+type CreateQuoteMetricParam = {
+  sessionData: SessionData
+  sendToSalesRep: boolean
+  userData: UserData
+  quoteId: string
+  quoteReferenceName: string
+  creationDate: string
+}
+
+type CreateQuoteFieldsMetric = {
+  cost_center_id: string
```

Comment:
Why did you create with different patterns, cameCase and snake_case?

Reply:
snake_case is the Analytics pattern. If you don't use it, we have to convert when send to them. I preferred to make the call simpler instead of converting them.

Reply:
There is a possibility of using `@JsonProperty`. I don't know what you think about it and if the IO support.

Reply:
I'm not aware of that. Is it a npm package?

Reply:
It is to change the property name on serialize or deserialize. For example, with [transcript-json-serialize](https://www.npmjs.com/package/typescript-json-serializer?activeTab=readme), it is just to keep our naming conventions in code and change the name in Decorator. What do you think?

Reply:
It's an interesting idea, but decorators are not frequently used in JS world. I didn't even know Typescript supported them at all. I did a quick search and it seems like on the typescript version we're stuck on now the decorators implementation is experimental (which is why the package docs requires enabling the `experimentalDecorators` flag). I don't think it's worth using the experimental feature, which is not yet commonly used in this stack yet.

Once we can upgrade IO to use Typescript 5 the decorator implementation will not be experimental anymore, so it might be worth trying it out. But we're still stuck on v3.

Reply:
I agree with @mairatma on this